### DISCUSSION TITLE
filter.conf words deter option

### DISCUSSION THREAD

**Discussion by legendskyk** [inspircd/inspircd#2005] [Support]
Hello,
Its possible to make deter word system in m_filter module?
like if someone using bad word he think his msgs appear to on main but thats not sent to main rather it preserved in logs file in deter.txt?

**Comment by sadiepowell** [comment]
You can do this using a `<log>` tag with the type `m_filter FILTER REMOTEFILTER`, set `<filteropts notifyuser="no">`, and then use the `block` filter type.

https://docs.inspircd.org/3/configuration/#log

https://docs.inspircd.org/3/modules/filter/#filteropts

**Comment by legendskyk** [comment]
no i dont meant that
I mean a function like already filter have like:

	FA_GLINE,
	FA_BLOCK,
	FA_SILENT,
	FA_KILL,
	FA_NONE
and you added more a option like:
       FA_DETER,

I want a function in that module by deter mean if we add a word like for example "sex" in filter.conf like this:

keyword pattern="sex" action="deter" duration="3d" reason="filter 1" flags="*" 

then that word not appear chat rather its hidden and save into logs folder by name deter.txt
user not get any warning he think his msgs being sent but in real that msg being filter to deter and not appear to chat even in main also pm.

**Comment by sadiepowell** [comment]
What I mentioned will do that other than it will only log the filter which was matched rather than the full message which is intentionally not logged to avoid networks snooping on private messages.

**Comment by legendskyk** [comment]
It will not appear on chat room too?? if we fitler out the words?

**Comment by sadiepowell** [comment]
If you filter a message with `<filteropts notifyuser="no">` set then the user will have the original message shown to them.

**Comment by legendskyk** [comment]
shown to them but no showing to the others right?

**Comment by sadiepowell** [comment]
Yes? Why would it be called "block" if it didn't block the message?

**Comment by legendskyk** [comment]
ok thats what i want let me try to it.
